src/express.ts

```diff
@@ -387,12 +397,21 @@ export class InngestCommHandler {
       console.warn("Couldn't unpack register response:", err);
     }
     const { status, error } = registerResSchema.parse(data);
-    console.log(
-      "Registered Inngest functions:",
-      res.status,
-      res.statusText,
-      data
-    );
+
+    // The dev server polls this endpoint to register functions every few
+    // seconds, but we only want to log that we've registered functions if
+    // the function definitions change.  Therefore, we compare the body sent
+    // during registration with the body of the current functions and refuse
+    // to register if the functions are the same.
+    if (lastRegisteredHash !== body.hash) {
```

Comment:
I'm not sure this would work in a serverless environment where this file will be imported and run on every request. I'm not sure it'd be a problem for Express if it's a running process, but the other handlers would have a more difficult time.

Instead of comparing the hash here, can we have the dev server do the work and return a particular piece of data stating that no change has been made? It has a much better chance of being stateful than this code.

Reply:
You're right about serverless or a server that is restarted often during dev (think: [nodemon](https://nodemon.io/)). I think with Tony's dev server change (https://github.com/inngest/inngest/pull/263) it does do that as well so it is kind of a progressive enhancement on both sides of the system. When we add deployments to Inngest Cloud, we can emulate what the dev server is doing as well.

Reply:
We'll handle 202s && "skipped: true" as if the functions were already registered.